### PR TITLE
PR: merge working branch 4

### PR DESCRIPTION
**Pull request by pitzelalex**

### CODE REVIEW

File: lib/game.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def col_select
       col = convert_input(input)
       if !["a", "b", "c", "d", "e", "f", "g"].include? input.downcase
```
Comment (pitzelalex) [inline lib/game.rb:23]: Curious about these ! bangs

Comment (dlayton66) [reply lib/game.rb:23]: I wanted to ask, "is this array MISSING this element?"  I tried finding an `<array>.exclude?` method, but it looks like that doesn't exist in Ruby by default so I went with `!<array>.include?`

File: lib/board.rb [hunk 2]
```diff
@@ -12,6 +12,15 @@ def initialize
       row_6: ".......",
       }
     @open_row = [6,6,6,6,6,6,6]
-    @playable_col = [0,1,2,3,4,5,6]
+    @open_col = [0,1,2,3,4,5,6]
```
Comment (dlayton66) [inline lib/board.rb:15]: I like both the grid and open_col refactors.  Makes the code easier to write and follow.

File: lib/board.rb [hunk 3]
```diff
@@ -12,6 +12,15 @@ def initialize
...
+  end
+
+  def game_over?
+    @open_col == [] # || four of a kind win con check goes here
+  end
+
+  def update_open(col)
```
Comment (dlayton66) [inline lib/board.rb:22]: Very slick.

**Review by dlayton66 — APPROVED** [review]
Great additions all around.

**Pull request merged**